lib/rialto/etl/service_client/web_of_science_client.rb

```diff
@@ -13,34 +13,43 @@ class WebOfScienceClient
         DEFAULT_QUERY_ID = 0
         NO_RECORDS_FOUND = 0
 
-        def initialize(firstname:, lastname:, institution:)
-          @lastname = lastname
-          @firstname = firstname
+        def initialize(institution:)
           @institution = institution
         end
 
-        attr_reader :lastname, :firstname, :institution
+        attr_reader :institution
 
         # Hit the API endpoint and iterate over resulting records
         def each
           return to_enum(:each) unless block_given?
 
-          # Run the initial query to get a query ID (for pagination)
-          perform_initial_query!
+          publication_ranges.each do |publication_range|
+            self.publication_range = publication_range
 
-          records = []
-          first_record_values.each do |first_record|
-            records += query_by_id(first_record: first_record)
-          end
-          # Yield the block for each result on the page
-          records.each do |val|
-            yield val
+            # Run the initial query to get a query ID (for pagination)
+            perform_initial_query!
+
+            logger.info "records found: #{records_found}"
+
+            first_record_values.each do |first_record|
+              yield query_by_id(first_record: first_record)
+            end
           end
         end
 
         private
 
-        attr_reader :records_found, :query_id
+        attr_accessor :records_found, :query_id, :publication_range
+
+        def publication_ranges
```

Comment:
Eventually, yes. The current ranges have us covered until the end of 2021. This is admittedly suboptimal, but I'm not sure how else to use the WoS API reliably subdivide the Stanford pubs in that data set. We could add more future-based ranges, or extend that last futuristic one.

My hope is that the WoS REST API becomes more robust in the meantime so that we can eventually (perhaps in a second RIALTO work cycle?) ditch this approach altogether.